go/libraries/doltcore/env/actions/infer_schema.go

```diff
@@ -224,6 +224,12 @@ func leastPermissiveNumericType(strVal string, floatThreshold float64) (ti typei
 		if err != nil {
 			return typeinfo.UnknownType
 		}
+
+		// handle leading zero case
+		if strVal[0] == '0' {
+			return typeinfo.StringDefaultType
```

Comment:
@bheni returning string here since returning `typeinfo.UnknownType` will cause `strVal` to fall through to the chrono type https://github.com/dolthub/dolt/blob/master/go/libraries/doltcore/env/actions/infer_schema.go#L157

Reply:
I think you need to check for a following decimal point.  Like 0.5 should be a float.  I'd add that, and then add a test.